lte/protos/pipelined.proto

```diff
@@ -198,6 +198,217 @@ message PacketDropTableId {
   int32 table_id = 1;
 }
 
+//-------------------------------------------------------------------------
+// SMF to UPF Proto
+//-------------------------------------------------------------------------
+
+
+
+message NodeID {
+  enum NodeIDType {
+    IPv4 = 0;
+    IPv6 = 1;
+    FQDN = 2;
+    }
+  NodeIDType node_id_type = 1;
+  string node_id = 2;
+}
+
+
+//CPFseid message
+message CpFSeid {
+  uint64 f_seid = 1;
+  NodeID node_id = 2;
+}
+
+message OuterHeadRemoval {                //conditional PDR Message Structures
+  uint32 o_hd_remo_descr = 1;
+}
+
+
+//message LocalFTeid {                      //not sure about the message structure
+ ///local_f_teid = ;                        //doubt
+//}
+
+
+
+message SdfFilters {                               //8.2.5
+
+  FlowDescriptor fd = 1;
+  bytes ttc = 2;
+  bytes spi = 3;
+  bytes fl = 4;
+  uint32 sdf_id = 5;
+}
+
+message FlowDescriptor {                //include
+FlowMatchNew match = 1;
+  enum Action {
+    PERMIT = 0;
+    DENY = 1;
+  }
+  Action action = 2;
+}
+
+message FlowMatchNew {
+  string ipv4_src = 1;
+  string ipv4_dst = 2;
+  uint32 tcp_src = 3;
+  uint32 tcp_dst = 4;
+  uint32 udp_src = 5;
+  uint32 udp_dst = 6;
+}
```

Comment:
You may consider Oneof type here: https://developers.google.com/protocol-buffers/docs/proto3#oneof

Reply:
Thanks, I'll check and modify.